include/zephyr/dt-bindings/misc/nordic-domain-id-nrf54h20.h

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_DT_BINDINGS_MISC_NORDIC_DOMAIN_ID_NRF54H20_H_
+
+#define NRF_DOMAIN_ID_APPLICATION 2
+#define NRF_DOMAIN_ID_RADIOCORE   3
+#define NRF_DOMAIN_ID_GLOBALFAST  12
+#define NRF_DOMAIN_ID_GLOBALSLOW  13
+
+#ifndef __DTS__
+
+#include <nrf.h>
+#include <zephyr/toolchain.h>
+
+/* Check the above definitions against C enum from nRF MDK. */
+BUILD_ASSERT(NRF_DOMAIN_ID_APPLICATION == NRF_DOMAIN_APPLICATION);
+BUILD_ASSERT(NRF_DOMAIN_ID_RADIOCORE   == NRF_DOMAIN_RADIOCORE);
+BUILD_ASSERT(NRF_DOMAIN_ID_GLOBALFAST  == NRF_DOMAIN_GLOBALFAST);
+BUILD_ASSERT(NRF_DOMAIN_ID_GLOBALSLOW  == NRF_DOMAIN_GLOBALSLOW);
+
+#endif /* __DTS__ */
```

Comment:
build assertions should go into C files, not headers.

Reply:
also, we're polluting binding headers with validation stuff, these are independent things. this can be appended in the C file we already have to validate stuff (or add an additional one).

Reply:
> build assertions should go into C files, not headers.

There are lots of build assertions in Zephyr headers.

> Also, we're polluting binding headers with validation stuff, these are independent things. This can be appended to the C file we already have to validate stuff (or add another one).

That is a downside, but this solution is more scalable than having an additional C file, separate from the original definitions, with conditional includes for each SoC, to be maintained in tandem with these headers and the SoC dtsi.

Reply:
> > build assertions should go into C files, not headers.
> 
> There are lots of build assertions in Zephyr headers.
> 

Then they should be removed. Otherwise, you can easily pollute the build with N assertions when you typically want/need 1.

> > Also, we're polluting binding headers with validation stuff, these are independent things. This can be appended to the C file we already have to validate stuff (or add another one).
> 
> That is a downside, but this solution is more scalable than having an additional C file, separate from the original definitions, with conditional includes for each SoC, to be maintained in tandem with these headers and the SoC dtsi.

We already validate DT addresses in a single place. Why not a "validate_bindings_headers.c", always included in nRF SoCs builds?

Reply:
I am in favor of the `.c` file as well, this is what we do in Bluetooth too:

https://github.com/zephyrproject-rtos/zephyr/blob/f4c067c6bcf0fa63b2f52d3dfb4879d73815b9fc/subsys/bluetooth/common/dummy.c#L22